### PR TITLE
[Linux] Fix cross-architecture component build configuration for ARM32-target

### PR DESCRIPTION
**Pull request by hseok-oh**
Fix CMake install function: pair `_install` with `add_library_clr`
related issue: #8685 - warning target `palrt`

### CODE REVIEW

**Comment by jkotas** [comment]
cc @janvorli

**Comment by janvorli** [comment]
I am not sure how this change would fix the dependency issue. This only changes how the final version of the static palrt.lib is copied to the final Product folder and  while before it was installed there for all builds, now it is installed only when the CLR_CROSS_COMPONENTS_BUILD is not defined.
The messages from the #8685 rather indicate that the coreclrpal is not being built for the cross arch component

**Comment by hseok-oh** [comment]
@janvorli You're right. It only fix warning about installation palrt. We need to modify CMakeLists.txt on project root becase coreclrpal is not being built as you say. I'll push additional commit in this PR.
https://github.com/dotnet/coreclr/blob/master/CMakeLists.txt#L551-L558
And if we fix this, we will be meet many build error, as next step.

**Comment by hseok-oh** [comment]
additional commit

- modify CMakeLists.txt to add `coreclrpal` for cross-architecture component
- fix build.sh and  CMakeLists.txt to allow only x64-host/arm-target cross-architecture component for linux: `coreclrpal` for linux x86 is not stable now

File: CMakeLists.txt [hunk 1]
```diff
@@ -129,14 +129,8 @@ if(CMAKE_SYSTEM_NAME STREQUAL Linux)
...
-        if(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL x86_64 OR CMAKE_SYSTEM_PROCESSOR STREQUAL amd64)
-            if(CLR_CMAKE_TARGET_ARCH STREQUAL "arm")
-                set(CLR_CMAKE_PLATFORM_UNIX_X86 1)
-            else()
-                set(CLR_CMAKE_PLATFORM_UNIX_AMD64 1)
-            endif()
-        elseif(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL i686)
-            set(CLR_CMAKE_PLATFORM_UNIX_X86 1)
+        if(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL x86_64 OR CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL amd64)
+            set(CLR_CMAKE_PLATFORM_UNIX_AMD64 1)
```
Comment (janvorli) [inline CMakeLists.txt]: I believe that at the moment, 64 bit cross components build will not work for 32 bit target and so the x86 host is the only option for target ARM.  See the comment from @AndyAyersMS here:
https://github.com/dotnet/corert/issues/2381#issuecomment-268010021

Comment (AndyAyersMS) [inline CMakeLists.txt]: The JIT may actually build, but it probably won't run correctly. Last I knew there were still parts of the jit code that assumed sizeof(void *) was the target pointer size.

**Comment by hseok-oh** [comment]
@janvorli PTAL

**Pull request merged**